Automattic/wp-calypso
# [Woo CYS] Implement `Appearance > Themes` entry point

Related to #81993 

## Proposed Changes

* Implement Customize Your Store  Appearance > Themes entry point

figma: Mrk6SERPZ4KrFHSjM0a8TK-fi-3388%3A275564

```
Feature: Customize Your Store Intro page when the task is completed

  Scenario: use case 1
    Given user Not yet created a store with AI and the assembler, No changes to any theme (Tsubaki is still the active theme)
    Then it should: 
     1. Display "The Design with AI component" right after the first row.
     2. Remove the “Install theme” button” on the top right.
     3. Align the Upgrade banner with the active color scheme.

  Scenario: use case 2
    Given Not yet created a store with AI and the assembler, Changes to the active theme
    Then it should: 
     1. Display "The Design with AI component" right after the first row.
     2. Display a modal to warn the user we’ll replace content.


  Scenario: use case 2
    Given Already created a store with AI and the assembler, and custom theme is the active theme
    Then it should: 
     1. Show custom theme thumbnail and actions
     2.  Display "The Design with AI component" at the bottom
```
<img width="1320" alt="theme-showcase" src="https://github.com/Automattic/wp-calypso/assets/4344253/7381afcd-2896-42ed-9fd7-3e745f30e885">


![Screenshot 2023-12-06 at 12 05 43](https://github.com/Automattic/wp-calypso/assets/4344253/8a6d51bd-124b-49a0-85ed-a140affed0b1)

![Screenshot 2023-12-06 at 11 28 37](https://github.com/Automattic/wp-calypso/assets/4344253/4548878d-e10f-46a2-8353-f82cf0d0018b)


## Testing Instructions



* Creating a Woo Express Site for Testing CYS pdibGW-2pG-p2
* Change [this](https://github.com/Automattic/wp-calypso/pull/84894/commits/af5d29122f781388fb8e6c2204112e6665c80456#diff-ff4b4ca8173b5696121aac959f8fde70b40dabc0306ef0adfc5f82cd328832faR20) line to return true
* Go to `http://calypso.localhost:3000/themes/<site-url>`
* Confirm 
    1. "The Design with AI component" is displayed right after the first row.
    2. “Install theme” button” on the top right is hidden.
    3.  "Upgrade now" or "Let's get started" Banner should align with the active color scheme.
![Screenshot 2023-12-06 at 13 41 56](https://github.com/Automattic/wp-calypso/assets/4344253/3adfa5a6-d1e3-419a-922d-4efdfe0bd9db)

* Go to `Tools > WCA Test Helper > Features` and enable `customize-store` feature flag.
* Click on `Design with AI` 
* You should be redirected to /wp-admin/admin.php?page=wc-admin&path=%2Fcustomize-store%2Fdesign-with-ai
* Change your theme
* Click on `Design with AI` 
* A "Start new design" warning model should be displayed at the start.
* Go to WooCommerce > Home
* Completed Customize Your Store task
* Go to `http://calypso.localhost:3000/themes/<site-url>`
* It should show a custom theme thumbnail and actions
* it should display "The Design with AI component" at the bottom
* Please test around 

## Pre-merge Checklist
